### PR TITLE
Challenge 2 - Final 

### PR DESCRIPTION
**Pull request by giovponce**
## Description
```
< Capstone 2, Final Deliverable>

- When the user enters the app, the app should show the Picture of the Day.
- When the user selects a specific date with the format YYYY-MM-DD, the app should show the picture of the day for the given date.
- The app should be responsive.
- When the app fetches the API, and there is an unexpected error, the app should show a message: "An error ocurred while retrieving data, please try again."
- When the user selects an invalid date value and clicks on the show button, the app should show a message error from the API response.
- Tests added.
```

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update
- [x] Tests (Adding missing tests or correcting existing tests)
- [x] Styles (Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc))

## What is the deliverable?
Capstone 2, final deliverable

## Screenshots and links

Screen Shot 2022-01-25 at 11 52 19
Screen Shot 2022-01-25 at 12 01 18
Screen Shot 2022-01-25 at 12 02 15
Screen Shot 2022-01-25 at 12 03 24

## Notes
Thanks again for everything, Sergio.

## Checklist
- [x] I added the necessary documentation, if appropriate.
- [x] I added tests to prove that my fix is effective or my deliverable works.
- [x] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

File: src/Pages/CustomDatePhoto.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import { StyledImgContainer, StyledImg, StyledImgText } from "../Utils/Styled Components/StyledImg";
+import { StyledIFrame } from "../Utils/Styled Components/StyledIFrame";
+
+export default function CustomDatePhoto() {
+
+    let initDate = {
+        target: {
+            value: "2022-01-19"
+        }
+    }
```
Comment (sergio-vf) [inline src/Pages/CustomDatePhoto.js]: Why don't you store value directly? 
```suggestion
    const initDate = "2022-01-19";
```

File: src/Pages/CustomDatePhoto.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    const [date, setDate] = useState(initDate);
+    const [data, setData] = useState(null);
+    const [video, setVideo] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState(false);
```
Comment (sergio-vf) [inline src/Pages/CustomDatePhoto.js]: You can use useReducer to handle this process

File: src/Pages/CustomDatePhoto.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                <StyledInput type="date" id="date" data-testid="date-input" value={date.target.value} selected={date} onChange={preSetDate} />
+                <br/><br/>
+                <StyledImgContainer>
+                    <StyledImg data-testid="img" src={data?.url}></StyledImg>
+                    <StyledImgText>{data?.copyright} ®</StyledImgText>
+                </StyledImgContainer>
+                <StyledTitle>{data?.title}</StyledTitle>
+                <StyledDescription>{data?.explanation}</StyledDescription>
+            </div>
+        )}
```
Comment (sergio-vf) [inline src/Pages/CustomDatePhoto.js]: You can delete some duplicity
```suggestion
     
            <div>
                <StyledInput type="date" id="date" data-testid="date-input" value={date.target.value} selected={date} onChange={preSetDate} />
                <br/><br/>
                <StyledImgContainer>
                video ?
                    <StyledIFrame data-testid="img" title={data.title} src={data.url}></StyledIFrame>
                    :
                    <StyledImg data-testid="img" src={data?.url}></StyledImg>
                    <StyledImgText>{data?.copyright} ®</StyledImgText>
                </StyledImgContainer>
                <StyledTitle>{data?.title}</StyledTitle>
                <StyledDescription>{data?.explanation}</StyledDescription>
            </div>
        ) 
```

File: src/Pages/PhotoOfTheDay.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
```
Comment (sergio-vf) [inline src/Pages/PhotoOfTheDay.js]: Probably you can add a  prop to this component that you can use to show/hide datepicker, so this way you just need 1 component for pickoftheday and customdate

File: src/Pages/RandomPhoto.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from "react";
```
Comment (sergio-vf) [inline src/Pages/RandomPhoto.js]: This component is also pretty similar to the others, i think you can make 1 generic component that handles your 3 useCases

File: src/Tests/CustomDatePhoto.test.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            expect(img).toBeInTheDocument();
+        });
+    });
+
+    test('changes the input', async () => {
+        render(<CustomDatePhoto />);
+        await waitFor(() => {
+            const input = screen.getByTestId('date-input');
+            fireEvent.change(input, { target: { value: '2022-01-20' } }); //eslint-disable-line
+            expect(input).toBeInTheDocument();
```
Comment (sergio-vf) [inline src/Tests/CustomDatePhoto.test.js]: I think a better assertion would be to check for the new value
```suggestion
            expect(input.value).toBe('2022-010-20');
```

File: src/Utils/Styled Components/StyledHamburguer.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+    ${StyledSpanForHamburguer}:first-child{
+      transform-origin: 0% 0%;
+    }
+
+    ${StyledSpanForHamburguer}:nth-last-child(2) {
+      transform-origin: 0% 100%;
+    }
+
+    ${StyledInputForHamburguer}:checked ~ ${StyledSpanForHamburguer} {
```
Comment (sergio-vf) [inline src/Utils/Styled Components/StyledHamburguer.js:99]: nice!! you are getting expert on this

**Review by sergio-vf — COMMENTED** [review]
I think in general you did a good job, just try to avoid code duplicity

File: src/Utils/Styled Components/StyledHamburguer.js (hunk 7 above)
Comment (giovponce) [reply src/Utils/Styled Components/StyledHamburguer.js:99]: Thanks! :3

**Comment by giovponce** [comment]
Applied suggestions, merging PR

**Pull request merged**